config/src/main/java/com/quorum/tessera/config/constraints/ServerConfigsValidator.java

```diff
@@ -1,45 +1,57 @@
+public class ServerConfigsValidator implements ConstraintValidator<ValidServerConfigs, Config> {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ServerConfigsValidator.class);
 
     @Override
-    public boolean isValid(List<ServerConfig> serverConfigs, ConstraintValidatorContext constraintContext) {
-        if(serverConfigs == null) {
+    public boolean isValid(Config config, ConstraintValidatorContext constraintContext) {
+        if (config == null || config.getServerConfigs() == null || config.getServerConfigs().isEmpty()) {
             return true;
         }
 
-        final Map<AppType,Integer> counts = serverConfigs.stream()
-            .collect(Collectors.toMap(ServerConfig::getApp,v -> 1,(l,r) -> l + 1));
+        final Map<AppType, Integer> counts =
+                config.getServerConfigs().stream()
+                        .collect(Collectors.toMap(ServerConfig::getApp, v -> 1, (l, r) -> l + 1));
 
-        final int p2PEnabledConfigsCount = counts.getOrDefault(AppType.P2P,0);
-        final int q2TEnabledConfigsCount = counts.getOrDefault(AppType.Q2T,0);
+        final int p2PEnabledConfigsCount = counts.getOrDefault(AppType.P2P, 0);
+        final int q2TEnabledConfigsCount = counts.getOrDefault(AppType.Q2T, 0);
 
         if (p2PEnabledConfigsCount != 1) {
             LOGGER.debug("Only one P2P server must be configured and enabled.");
             constraintContext.disableDefaultConstraintViolation();
-            constraintContext.buildConstraintViolationWithTemplate("Only one P2P server must be configured and enabled.")
-                .addConstraintViolation();
+            constraintContext
+                    .buildConstraintViolationWithTemplate("Only one P2P server must be configured and enabled.")
+                    .addConstraintViolation();
             return false;
         }
 
-        if (q2TEnabledConfigsCount == 0) {
+        if (config.isBootstrapNode()) {
+            if (q2TEnabledConfigsCount != 0) {
+                LOGGER.debug("Q2T server cannot be specified on a bootstrap node.");
+                constraintContext.disableDefaultConstraintViolation();
+                constraintContext
+                        .buildConstraintViolationWithTemplate("Q2T server cannot be specified on a bootstrap node.")
+                        .addConstraintViolation();
+                return false;
+            }
+        } else if (q2TEnabledConfigsCount == 0) {
             LOGGER.debug("At least one Q2T server must be configured and enabled.");
             constraintContext.disableDefaultConstraintViolation();
-            constraintContext.buildConstraintViolationWithTemplate("At least one Q2T server must be configured and enabled.")
-                .addConstraintViolation();
+            constraintContext
+                    .buildConstraintViolationWithTemplate("At least one Q2T server must be configured and enabled.")
```

Comment:
`and enabled` I think can be removed since we removed the ability for a server to be in a disabled state.

Reply:
I think we should add `or bootstrap mode enabled` to hint the user as to what they should do if they meant to have no Q2T servers.

Reply:
Good points

Reply:
Same for log msg on line 50